### ISSUE TITLE
On AL2023, q doctor claims: The /etc/ssh/sshd_config file needs to have the following line: 

### ISSUE THREAD

**Issue by rstrahan** [aws/amazon-q-developer-cli#302]
### Checks

- [X] I have searched [github.com/aws/amazon-q-developer-cli/issues](https://github.com/aws/amazon-q-developer-cli/issues?q=) and there are no duplicates of my issue
- [X] I have run `q doctor` in the affected terminal session
- [X] I have run `q restart` and replicated the issue again

### Operating system

Linux 6.1.112-124.190.amzn2023.x86_64 - Amazon Linux 2023.6.20241212

### Expected behaviour

I did install q commandline for remote AL2023 per docs.
I sucessfully configured it for PRO licence and logged in,
I did add the required lines to /etc/ssh/sshd_config
I did restart sshd
I did disconnect and log in again
I expect q doctor to report that all is well.

I tried to run q restart - it says  "error: Please restart Amazon Q from your host machine" but q is not installed on my local machine (bot supported on Windows)

### Actual behaviour

q commandline doesn't work and q doctor reports this error.

### Environment

```yaml
[q-details]
version = "1.5.1"
hash = "43f92e418e96c7beaa3e881f299241c15ba65900"
date = "2024-12-13T00:51:07.605173Z (9d ago)"
variant = "minimal"

[system-info]
chip = "Intel(R) Xeon(R) CPU E5-2686 v4 @ 2.30GHz"
total-cores = 4
memory = "15.61 GB"

[system-info.os.linux]
kernel_version = "6.1.112-124.190.amzn2023.x86_64"
id = "amzn"
name = "Amazon Linux"
pretty_name = "Amazon Linux 2023.6.20241212"
version_id = "2023"
version = "2023"

[environment]
cwd = "/home/USER"
cli-path = "/home/USER"
os = "Linux"
shell-path = "/usr/bin/bash"
shell-version = "5.2.15"
install-method = "unknown"
in-ssh = true

[env-vars]
PATH = "/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:/home/USER/miniconda3/bin:/home/USER/miniconda3/condabin:/home/USER/.console-ninja/.bin:/home/USER/.nvm/versions/node/v18.20.4/bin:/home/USER/.local/bin:/home/USER/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/home/USER/.local/bin"
Q_SET_PARENT_CHECK = "1"
SHELL = "/bin/bash"
TERM = "xterm-256color"
XDG_SESSION_TYPE = "tty"
```

**Comment by grant0417** [comment]
Make sure you have installed the ssh integrations on the host machine, you need to run `q integrations install ssh` on you macOS machine.

**Comment by rstrahan** [comment]
I don’t have a MacOS machine. I do have a local Windows machine. It did not appear possible to deploy  q integrations install ssh on Windows? Is there a path fwd for users of local Windows machines?

**Comment by grant0417** [comment]
I see, this error is a false positive then and you should be able to ignore it given everything else is working. If you run `q doctor --verbose` are there any other errors?

**Comment by rstrahan** [comment]
No - this is the only reported error.
```
$ q doctor --verbose
bash: __bp_last_argument_prev_command: parameter null or not set

✘ sshd config: Could not read sshd_config

  The /etc/ssh/sshd_config file needs to have the following line:
    AcceptEnv Q_SET_PARENT
    AllowStreamLocalForwarding yes
  
  If your sshd_config is already configured correctly then:
    1. Restart sshd, if using systemd: sudo systemctl restart sshd
    2. Disconnect from the remote host
    3. Run q integrations install ssh on the local machine
    4. Reconnect to the remote host and run q doctor again
  
  See https://docs.aws.amazon.com/amazonq/latest/qdeveloper-ug/command-line-autocomplete-ssh.html for more info


✘ Doctor found errors. Please fix them and try again.

If you are not sure how to fix it, please open an issue with q issue to let us know!
```

Looks like you are right, that it's a false positive, and I can ignore it..  Trying `q chat`, `q translate` etc.. seems all is well. 

This of course is not obvious, so hopefully you'll keep the issue open to fix it.. But in the meantime I'm unblocked - thanks so much!

**Comment by grant0417** [comment]
@brandonskiser

**Comment by JoanChirinos** [comment]
+1, same thing on Amazon Linux 2. All steps followed, error message doesn't go away but doesn't seem to be breaking anything

**Comment by YiWuManna** [comment]
+1 using Amazon Linux 2, follow all steps, but still report sshd_config error.

**Comment by scottschreckengaust** [comment]
Same and many others reporting. 

* https://github.com/aws/amazon-q-developer-cli/issues/3602
* https://github.com/aws/amazon-q-developer-cli/issues/2069
* https://github.com/aws/amazon-q-developer-cli/issues/2493
* https://github.com/aws/amazon-q-developer-cli/issues/2395
* https://github.com/aws/amazon-q-developer-cli/issues/3557
* https://github.com/aws/amazon-q-developer-cli/issues/3740

**Comment by scottschreckengaust** [comment]
I think the answer is simple, check if `sshd_config` is readable:

```shell
USER@remote % ls -al /etc/ssh/sshd_config
  -rw-------. 1 root root 3509 May 4 19:49 /etc/ssh/sshd_config
```

This permission is standard and intentional for security reasons:

  - Contains sensitive configuration — sshd_config controls how the SSH daemon behaves: which authentication methods are allowed, which ports to listen on, whether root login is permitted, allowed ciphers, etc. Exposing this to non-root users reveals your security posture.
  - Prevents information disclosure — An attacker with a low-privilege shell could read the config to learn things like whether password auth is enabled, what ports are open, or whether agent forwarding is allowed, then tailor their attack accordingly.
  - Default on many distros — RHEL/CentOS/Fedora (the . at the end of your permissions indicates SELinux context, so you're likely on one of these) ship sshd_config as 600 by default. Debian/Ubuntu often use 644, which is less restrictive but still common.

---

> [!WARNING]
> If you understand the consequences and need to have the doctor check the settings, updating the access is an option:
>
>```shell
>% sudo chmod go+r /etc/ssh/sshd_config
>```
